Documentation/cmdref/cilium-agent.md

```diff
@@ -49,6 +49,7 @@ cilium-agent [flags]
       --bpf-lb-maglev-table-size uint                             Maglev per service backend table size (parameter M) (default 16381)
       --bpf-lb-map-max int                                        Maximum number of entries in Cilium BPF lbmap (default 65536)
       --bpf-lb-mode string                                        BPF load balancing mode ("snat", "dsr", "hybrid") (default "snat")
+      --bpf-lb-no-control-plane                                   BPF load balancer uses an externally-provided control plane
```

Comment:
IMO it would be nicer if we could place this as part of the Hive Cell providing the K8sClient (and therefore ~ the K8s Controlplane integration) itself (https://github.com/cilium/cilium/blob/main/pkg/k8s/client/cell.go#L55) and naming it accordingly.

All the places that currently would check for the new flag could check whether the K8sClient is enabled or not.

But this change would only be possible if the K8sClient is available in all the places that check for the controlplane.